### PR TITLE
DSS-470: chore:(link) - Housekeeping

### PR DESCRIPTION
**Pull request by pixelflips**
# Description

Minor housekeeping-related adjustments to the link component including:

- Adding copy/descriptions for stories.
- Convert classes to function in order to prevent partial class names.
- Convert external icon to use `pds-icon`

Fixes [DSS-470](https://kajabi.atlassian.net/browse/DSS-470)

## Type of change

Please delete options that are not relevant.
If your type of change is not present, add that option.

- [x] Chore
- [x] This change requires a documentation update

# How Has This Been Tested?

1. Navigate to the link component and check impartial class names no longer appear in each story.
2. Check that the added docs descriptions make sense and are correct.

- [x] tested manually
- [ ] other:

**Test Configuration**:

- Pine versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

[DSS-470]: https://kajabi.atlassian.net/browse/DSS-470?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/core/src/components/pds-link/pds-link.tsx [hunk 1]
```diff
@@ -54,7 +54,12 @@ export class PdsLink {
     );
 
     return (
-      <a id={this.componentId} href={this.href} class={`pds-link pds-link--${this.variant} pds-link--${this.fontSize} `} target={this.external ? '_blank' : undefined}>
+      <a
+        id={this.componentId}
+        href={this.href}
+        class={`pds-link ${this.variant ? `pds-link--${this.variant}` : ''} ${this.fontSize ? `pds-link--${this.fontSize}` : ''}`}
```
Comment (ju-Skinner) [inline libs/core/src/components/pds-link/pds-link.tsx]: Can we extract the logic out of here to function like we do the other Components?

e.g 
```
const classNames = () => {
  const classes = ['pds-link']
  
  if ( this.variant) {
    classes.push(`pds-link--${this.variant}`);
  }
  
  if (this.fontSize) {
    classes.push(`pds-link--${this.fontSize}`);
  }
}
```

**Review by ju-Skinner — CHANGES_REQUESTED** [review]
Looks good. I just ask for one small change.

**Comment by pixelflips** [comment]
Added one additional update to swap out the hard-coded SVG to use `pds-icon` along with updating the applicable tests.

File: libs/core/src/components/pds-link/pds-link.tsx (hunk 1 above)
Comment (pixelflips) [reply libs/core/src/components/pds-link/pds-link.tsx]: Done. Great catch!

**Review by anechol — APPROVED** [review]
🚀

**Review by QuintonJason — APPROVED** [review]

**Pull request merged**